### PR TITLE
Task/deselect all orgunits

### PR DESCRIPTION
**Pull request by neeilya**
Fixes DHIS2-5260.

Changes proposed in this pull request:
- Enable support for deselecting all org units

### CODE REVIEW

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 1]
```diff
@@ -43,6 +44,7 @@ class OrgUnitSelector extends Component {
                     counter += 1;
 
                     if (counter > 1) {
+                        // eslint-disable-next-line
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: Is there a danger of infinite updating with setState in CDU?

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 2]
```diff
@@ -99,49 +101,70 @@ class OrgUnitSelector extends Component {
...
+                {this.props.selected.length > 0 && (
+                    <div style={styles.footer.tooltip}>
+                        {this.props.selected.length} selected.
+                        <a
+                            onClick={this.props.onDeselectAllClick}
+                            style={styles.footer.tooltip.link}
+                            // eslint-disable-next-line
+                            href="javascript:void(0)"
+                        >
+                            Deselect all
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: internationalization

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 3]
```diff
@@ -99,49 +101,70 @@ class OrgUnitSelector extends Component {
...
-                        displayEmpty
-                        fullWidth
-                    >
-                        {this.props.groupOptions.map(option => (
-                            <MenuItem key={option.id} value={option.id}>{option.displayName}</MenuItem>
-                        ))}
-                    </Select>
-                </FormControl>
+        <div style={styles.footer.index}>
+            <div style={styles.footer.tooltipContainer}>
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: Since this tooltip isn't an option and its being absolutely positioned, it seems to be logically belong better to the div that wraps the OrgUnitTree. I was also thinking that "tooltip" is a bit misleading as a term. Maybe snackbar or something like that?

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 4]
```diff
@@ -99,49 +101,70 @@ class OrgUnitSelector extends Component {
...
+                            // eslint-disable-next-line
+                            href="javascript:void(0)"
+                        >
+                            Deselect all
+                        </a>
+                    </div>
+                )}
+            </div>
+            <Grid style={styles.footer.gridContainer} container>
+                <Grid
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: Even though this code was not part of this PR, you could remove some code duplication (of the <Grid components) by adding a stateless component like so (not sure if GridControl is the best name):
```
<GridControl
                    title={i18n.t('Level')}
                    id="level-select"
                    value={this.props.level}
                    onChange={this.props.onLevelChange}
                    options={this.props.levelOptions}
                    disabled={this.props.userOrgUnits.length > 0}
                    renderValue={this.renderLevelOptions}
                    multiple
                />```

And 
```
const GridControl = ({id, title, options, ...props}) => {
    return (<Grid
        item
        xs={4}
        style={styles.footer.gridContainer.gridItem}
    >
        <FormControl style={{ width: '100%' }}>
            <InputLabel htmlFor={id}>{title}</InputLabel>
            <Select
                input={<Input id={id} />}
                fullWidth
                {...props}
            >
                {options.map(option => (
                    <MenuItem key={option.id} value={option.id}>{option.displayName}</MenuItem>
                ))}
            </Select>
        </FormControl>
    </Grid>);
}
```

Comment (jenniferarnesen) [reply packages/org-unit-dialog/src/OrgUnitSelector.js]: And the component (might not display correctly here):

``````
const GridControl = ({id, title, options, ...props}) => {
    return (<Grid
        item
        xs={4}
        style={styles.footer.gridContainer.gridItem}
    >
        <FormControl style={{ width: '100%' }}>
            <InputLabel htmlFor={id}>{title}</InputLabel>
            <Select
                input={<Input id={id} />}
                fullWidth
                {...props}
            >
                {options.map(option => (
                    <MenuItem key={option.id} value={option.id}>{option.displayName}</MenuItem>
                ))}
            </Select>
        </FormControl>
    </Grid>);
}```

File: packages/org-unit-dialog/src/OrgUnitSelector.js [hunk 5]
```diff
@@ -99,49 +101,70 @@ class OrgUnitSelector extends Component {
...
-                            <MenuItem key={option.id} value={option.id}>{option.displayName}</MenuItem>
-                        ))}
-                    </Select>
-                </FormControl>
+        <div style={styles.footer.index}>
+            <div style={styles.footer.tooltipContainer}>
+                {this.props.selected.length > 0 && (
+                    <div style={styles.footer.tooltip}>
+                        {this.props.selected.length} selected.
+                        <a
```
Comment (jenniferarnesen) [inline packages/org-unit-dialog/src/OrgUnitSelector.js]: Suggest using a styled <button> instead - more semantically correct and no hacking needed.

File: packages/org-unit-dialog/src/OrgUnitSelector.js (hunk 1 above)
Comment (neeilya) [reply packages/org-unit-dialog/src/OrgUnitSelector.js]: Yes, unless "if" condition is implemented.

**Comment by neeilya** [comment]
@jenniferarnesen just pushed changes related to your comments. Thanks.
